cli/cmd/install.go

```diff
@@ -331,3 +348,69 @@ func readIntoBytes(filename string) ([]byte, error) {
 
 	return buf.Bytes(), nil
 }
+
+func globalConfig(options *installOptions) *config.GlobalConfig {
+	var identityContext *config.IdentityContext
+	if options.enableTLS() {
+		identityContext = &config.IdentityContext{}
+	}
+
+	return &config.GlobalConfig{
+		LinkerdNamespace: controlPlaneNamespace,
+		CniEnabled:       options.noInitContainer,
+		IdentityContext:  identityContext,
+	}
+}
+
+func proxyConfig(options *installOptions) *config.ProxyConfig {
+	ignoreInboundPorts := []*config.Port{}
+	for _, port := range options.ignoreInboundPorts {
+		ignoreInboundPorts = append(ignoreInboundPorts, &config.Port{Port: uint32(port)})
+	}
+
+	ignoreOutboundPorts := []*config.Port{}
+	for _, port := range options.ignoreOutboundPorts {
+		ignoreOutboundPorts = append(ignoreOutboundPorts, &config.Port{Port: uint32(port)})
+	}
+
+	return &config.ProxyConfig{
+		ProxyImage: &config.Image{
+			ImageName:  options.taggedProxyImage(),
+			PullPolicy: options.imagePullPolicy,
+			Registry:   options.dockerRegistry,
+		},
```

Comment:
This produces a config that looks like:

```
  "proxyImage": {
    "imageName": "gcr.io/linkerd-io/proxy:dev-47a1f2fb-kl",
    "pullPolicy": "IfNotPresent",
    "registry": "gcr.io/linkerd-io"
  },
```

My original thinking was that we would represent the image as:

```
  "proxyImage": {
    "imageName": "proxy",
    "pullPolicy": "IfNotPresent",
    "registry": "gcr.io/linkerd-io"
  },
```

No registry or version in the image name. The registry already exists as a separate field, and the version should be determined by the control plane components that load this config, rather than the CLI that's used to generate it.